core/src/main/java/org/apache/carbondata/scan/filter/FilterUtil.java

```diff
@@ -1426,4 +1423,25 @@ private static void getUnknownExpressionsList(Expression expression,
       getUnknownExpressionsList(child, lst);
     }
   }
+  /**
+   * This method will compare double values it will preserve
+   * the -0.0 and 0.0 equality as per == ,also preserve NaN equality check as per
+   * java.lang.Double.equals()
+   *
+   * @param d1 double value for equality check
+   * @param d2 double value for equality check
+   * @return boolean after comparing two double values.
+   */
+  public static int compare(Double d1, Double d2) {
+    if ((d1.doubleValue() == d2.doubleValue()) || (Double.isNaN(d1) && Double.isNaN(d2))) {
+      return 0;
+    }
+    if (d1 < d2) {
```

Comment:
one condition we can save in else, its fine i will update as per your comments. Thanks for reviewing.